feat: parse ConEmu OSC9;2

dmehala opened the pull request:
This PR implements support for the [ConEmu OSC9;2 escape sequence](https://conemu.github.io/en/AnsiEscapeCodes.html#OSC_Operating_system_commands).

| Sequence | Description |
| - | - |
ESC ] 9 ; 2 ; ”txt“ ST | Show GUI MessageBox ( txt ) for any purposes.

#3125

Commit:
feat: parse ConEmu OSC9;2

mitchellh commented on src/terminal/osc.zig:
```diff
@@ -1662,6 +1681,19 @@ test "OSC: show desktop notification with title" {
     try testing.expectEqualStrings(cmd.show_desktop_notification.body, "Body");
 }
 
+test "OSC: OSC9;2 conemu message box" {
```
Can you add the following tests:

- Blank message
- Spaces only message
- No trailing semicolon `OSC 9;2`

Commit:
code review

- Change show_message_box from struct to string.
- Add tests:
  - Blank message
  - Spaces only message
  - No trailing semicolon OSC 9;2
